mlflow/spark.py

```diff
@@ -909,6 +995,12 @@ def predict(
         """
         from pyspark.ml import PipelineModel
 
+        if _is_spark_connect_model(self.spark_model):
+            # Spark connect ML model directly appends prediction result column to input pandas
+            # dataframe. To make input dataframe intact, make a copy first.
+            pandas_df = pandas_df.copy(deep=False)
+            return self.spark_model.transform(pandas_df)["prediction"]
```

Comment:
Are we sure that after transformation the model uses 'prediction' as output column name?

Reply:
BTW should we move 'pandas_df.copy(deep=False)' inside spark? I feel like spark users won't remember to call '.copy' and it's quite complex if they always need to make a copy of the pandas dataframe.

Reply:
> Are we sure that after transformation the model uses 'prediction' as output column name?

No. 'prediction' is the default setting in pysparkML model, User might customize the output column,
existing pyspark.ml model has the same issue. We need to inspect nested model to get prediction column param. I will file a separate PR to address it together for pyspark.ml and pyspark.ml.connect models

> BTW should we move 'pandas_df.copy(deep=False)' inside spark? I feel like spark users won't remember to call '.copy' and it's quite complex if they always need to make a copy of the pandas dataframe.

https://github.com/apache/spark/pull/42887 is filed, but it has to go into spark 4.0

Reply:
Let's add a comment.

Reply:
Can we use `output_column = set(input_df.columns) - set(output_df.columns)`?

Reply:
I think what MLflow needs is a method that takes a pandas dataframe and returns prediction values. Can we add such a method?

Reply:
Do we want that for legacy SparkML as well to maintain consistency? If so, which seems logical, what are the broader implications for this departure from the SparkML standard APIs with .transform() chaining? 
Will this have an impact on pipeline child object references?

Reply:
> Can we use output_column = set(input_df.columns) - set(output_df.columns)?

There's some issues:
* some estimators generate multiple output columns (e.g. prediction, probability, rawPrediction)
* for pipeline, each stage will generate its own output columns the result dataframe contains all output columns from every stage.